napari/_qt/qt_viewer.py

```diff
@@ -1104,6 +1124,37 @@ def on_draw(self, event):
                     shape_threshold=self.canvas.size,
                 )
 
+    def on_fps_update(self, event):
+        image_layers = [
+            layer for layer in self.viewer.layers if isinstance(layer, Image)
+        ]
+
+        nodes = []
+        for image in image_layers:
+            visual = self.layer_to_visual[image]
+            nodes.append(visual._layer_node.get_node(3))
+
+        fps = event.fps
+        if fps < 30:
+            for node in nodes:
+                node.relative_step_size = min(node.relative_step_size * 2, 2)
+        if fps > 45:
+            for node in nodes:
+                node.relative_step_size = max(node.relative_step_size / 2, 0.1)
+
+    def redraw_at_higher_resolution(self, event=None):
+        image_layers = [
+            layer for layer in self.viewer.layers if isinstance(layer, Image)
+        ]
+
+        nodes = []
+        for image in image_layers:
+            visual = self.layer_to_visual[image]
+            nodes.append(visual._layer_node.get_node(3))
+        for node in nodes:
+            node.relative_step_size = 0.8
+        self.canvas.update()
```

Comment:
Does tis have to be separate like this? Would be great if we could simply fire the above method in  a way that results in high quality... I think it would work with the above suggestion.